### PR TITLE
Add seadex-genesis library recipe - Conan 2 ready

### PR DESCRIPTION
**Pull request by Seadex-GmbH**
Specify library name and version:  genesis/2.0

Author of the library.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Seadex-GmbH** [comment]
Is there something wrong with the recipe? If yes, how can I check, as details are not provided?

**Comment by czoido** [comment]
Thanks a lot for contributing this @Seadex-GmbH 
Sorry for the late response, and thanks you for your patience. We are a bit delayed with reviews but we are trying to speed things up a bit these days. We will start reviewing your PR today.

File: recipes/seadex-genesis/all/test_v1_package/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.15)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../test_package/
+                 ${CMAKE_CURRENT_BINARY_DIR}/test_package/)
```
Comment (czoido) [inline recipes/seadex-genesis/all/test_v1_package/CMakeLists.txt:8]:
```suggestion
                 ${CMAKE_CURRENT_BINARY_DIR}/test_package/)

```

File: recipes/seadex-genesis/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+cmake_minimum_required(VERSION 3.15)
+project(test_package LANGUAGES CXX)
+
+find_package(seadex-genesis REQUIRED CONFIG)
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} PUBLIC seadex-genesis::seadex-genesis)
+target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_17)
```
Comment (czoido) [inline recipes/seadex-genesis/all/test_package/CMakeLists.txt:7]:
```suggestion
target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_17)

```

File: recipes/seadex-genesis/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.options["fmt/*"].header_only = True
+        self.options["spdlog/*"].header_only = True
```
Comment (czoido) [inline recipes/seadex-genesis/all/conanfile.py:51]: Is this really needed? I could build the library removing these, also from seadex-essentials

Comment (Seadex-GmbH) [reply recipes/seadex-genesis/all/conanfile.py]: The options "header_only" for the dependencies removed.

**Review by czoido — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Pull request merged**